src/preprocess/acdc_data_split.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import csv
+import logging
+import random
+from collections.abc import Iterable
+from pathlib import Path
+
+
+def main(args):
+    # Randomly and evenly split the data into k folds based on the diseases.
+    data_paths = sorted((args.data_dir / 'training').glob('**/Info.cfg'))
+    group_dict = {
+        'NOR': [],  # healthy patients
+        'MINF': [],  # patients with previous myocardial infarction
+        'DCM': [],  # patients with dilated cardiomyopathy
+        'HCM': [],  # patients with an hypertrophic cardiomyopathy
+        'RV': []  # patients with abnormal right ventricle
+    }
+    for data_path in data_paths:
+        with open(data_path, 'r') as f:
+            for line in f:
+                key, value = line.rstrip('\n').split(': ')
+                if key == 'Group':
+                    group_dict[value].append(data_path.parent.resolve())
+                    break
+    _, groups = zip(*sorted(group_dict.items()))
+    random.seed(0)
+    groups = tuple(random.sample(group, k=len(group)) for group in groups)
```

Comment:
Does it aim to randomly shuffle the `data_paths` in each group?

Reply:
Yes it's.